roles/aws/aws_ec2_with_eip/tasks/main.yml

```diff
@@ -114,3 +114,23 @@
     - aws_ec2_with_eip.route_53.zone | length > 0
     - aws_ec2_with_eip.route_53.wildcard is defined
     - aws_ec2_with_eip.route_53.wildcard
+
+- name: Create EC2 autorecovery alarm.
+  include_role:
+    name: aws/aws_ec2_metric_alarm
+  vars:
+    aws_ec2_metric_alarm:
+      aws_profile: "{{ aws_ec2_with_eip.aws_profile }}"
+      region: "{{ aws_ec2_with_eip.region }}"
+      name: "{{ aws_ec2_with_eip.instance_name }}_auto_recovery"
+      description: "This metric auto recovers EC2 instances"
+      metric: "StatusCheckFailed_System"
+      namespace: "AWS/EC2"
+      statistic: "Maximum"
+      comparison: "GreaterThanOrEqualToThreshold"
+      threshold: 1
+      unit: "Count"
+      period: 60
+      evaluation_periods: 2
+      alarm_actions: ["arn:aws:automate:{{ aws_ec2_with_eip.region }}:ec2:recover"]
+      dimensions: { "InstanceId": "{{ aws_ec2_with_eip_instances.instances[0].instance_id }}" }
```

Comment:
Let's add a `aws_ec2_with_eip.autorecover` boolean that defaults to `true` and add a `when: aws_ec2_with_eip.autorecover` to this task.

Reply:
Done! :)